src/backend/fs.rs

```diff
@@ -68,6 +88,8 @@ pub async fn write_file<'a>(
         .try_collect()
         .await?;
 
+    let mime_type_value = shared_mime_type.lock().await;
```

Comment:
This is a terrible approach. Please refactor to not use locks and instead just get the mimetype before segmentation. This might require doing that in the http server stream loop.